src/lib/registration/widgets/registration_code.rb

```diff
@@ -73,13 +74,38 @@ def register
         Storage::InstallationOptions.instance.base_registered = true
       end
 
+      def validate
+        return error(_("System already registered.")) if Registration.is_registered?
```

Comment:
In case of already registered system we should simply skip registration and display just a warning.

But as user normally cannot go back after registration (AFAIK the installation is started even after failed registration) this check is actually irrelevant. But let's keep it just to be on the safe side...